### PR TITLE
Updates Response for APIs returning an array & the CLI commands using the APIs

### PR DESCRIPTION
**Pull request by SM43**
This updates the response of the Category & Resources APIs. Now, the object/array will be returned inside an object with a key `data` instead of only an object/array. so that later new fields can be added to the response
of an object without breaking the APIs.
for eg. metadata for pagination

Response:
```
{
   "data": {} | []
}

```

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/design/resource.go [hunk 1]
```diff
@@ -73,13 +71,11 @@ var _ = Service("resource", func() {
...
 			Attribute("limit", UInt, "Maximum number of resources to be returned", func() {
-				Default(100)
+				Default(1000)
 				Example("limit", 100)
 			})
 		})
-		Result(CollectionOf(Resource), func() {
-			View("withoutVersion")
-		})
+		Result(ResourceResult)
```
Comment (PuneetPunamiya) [inline api/design/resource.go:78]: Can't this be done like this
```
Result(func() {
     Attribute("data", CollectionOf(Resource), "List of resources", func() {
		View("withoutVersion")
	})
})
```

Comment (SM43) [reply api/design/resource.go:78]: yeah it can be this way also.

Comment (PuneetPunamiya) [reply api/design/resource.go:78]: So here we can avoid creating a new type in `type.go`

Comment (SM43) [reply api/design/resource.go:78]: it is used for multiple APIs. and later when we add `metadata`, it has to be added at one place instead of multiple

Comment (SM43) [reply api/design/resource.go:78]: thats why created a new type.

Comment (SM43) [reply api/design/resource.go:78]: Will that be a issue?

Comment (PuneetPunamiya) [reply api/design/resource.go:78]: Nope not any issue :+1:

**Review by sthaha — COMMENTED** [review]
/hold

File: api/pkg/cli/hub/get_resource.go [hunk 2]
```diff
@@ -39,6 +39,13 @@ type ResourceResult struct {
 	version string
 }
 
+// APIResponseForResource is the response of API when finding a resource
+type APIResponseForResource = rclient.ByCatalogKindNameResponseBody
```
Comment (sthaha) [inline api/pkg/cli/hub/get_resource.go:43]: How about `ResourceResponse`?

File: api/pkg/cli/hub/get_resource.go [hunk 3]
```diff
@@ -39,6 +39,13 @@ type ResourceResult struct {
...
+
+// APIResponseForResourceWithTheVersion is the response of API when finding a resource
+// with a specific version
+type APIResponseForResourceWithTheVersion = rclient.ByCatalogKindNameVersionResponseBody
```
Comment (sthaha) [inline api/pkg/cli/hub/get_resource.go:47]: how about `ResourceWithVersionResponse` ?

Comment (sthaha) [inline api/pkg/cli/hub/get_resource.go:47]: should we export these 🤔 ?

**Review by sthaha — COMMENTED** [review]
/lgtm

We should wait for PR to experimental UI before merging this

/hold

**Pull request merged**